src/QsCompiler/CompilationManager/TypeChecking.cs

```diff
@@ -2207,6 +2217,15 @@ internal static void UpdateTypeChecking(this FileContentManager file, Compilatio
                     : contentTokens);
 
                 diagnostics = QsCompilerError.RaiseOnFailure(() => RunTypeChecking(compilation, declarationTrees, CancellationToken.None), "error on running type checking");
+
+                var numRemovedCycleDiags = file.CurrentSemanticDiagnostics().Count(m => m.Code == ErrorCode.InvalidCyclicTypeParameterResolution.Code())
+                    - diagnostics.Count(m => m.Code == ErrorCode.InvalidCyclicTypeParameterResolution.Code());
+
+                if (numRemovedCycleDiags > 0 || diagnostics.Any(x => x.Source != file.FileName.Value))
+                {
+                    file.TriggerGlobalTypeChecking();
+                }
```

Comment:
Similarly to the code added to `RunTypeChecking`, it could be nice if this was factored out into a separate function since this one is already fairly long.

Reply:
done